packages/flame/test/components/mixins/has_paint_test.dart

```diff
@@ -37,68 +170,189 @@ void main() {
+
+        const newColor = Color(0xFF123456);
+        final newPaint = Paint()..color = newColor;
+        comp.setPaint('test', newPaint);
+
+        expect(comp.paintLayers[0].color, newColor);
+      },
+    );
+
+    test(
+      'setPaint does not change paintLayers when updatePaintLayers is false',
+      () {
+        final comp = _MyComponent();
+
+        const testColor = Color(0xFFE5E5E5);
+        final testPaint = Paint()..color = testColor;
+        comp.setPaint('test', testPaint);
+        comp.setPaintLayers(['test']);
+
+        final newPaint = Paint()..color = const Color(0xFF123456);
+        comp.setPaint('test', newPaint, updatePaintLayers: false);
+
+        expect(comp.paintLayers[0].color, testColor);
+      },
+    );
+
+    test(
+      'deletePaint deletes paint from paintLayers by default',
+      () {
+        final comp = _MyComponent();
+
+        const color = Color(0xFFE5E5E5);
+        comp.setPaint('test', Paint()..color = color);
+        comp.addPaintLayer('test');
+        comp.deletePaint('test');
+
+        expect(comp.paintLayers.length, 1);
+      },
+    );
+
+    test(
+      'deletePaint does not delete paint from paintLayers when '
+      'updatePaintLayers is false',
+      () {
+        final comp = _MyComponent();
+
+        const color = Color(0xFFE5E5E5);
+        comp.setPaint('test', Paint()..color = color);
+        comp.addPaintLayer('test');
+        comp.deletePaint('test', updatePaintLayers: false);
+
+        expect(comp.paintLayers.length, 2);
+      },
+    );
+
+    test(
+      'delete paint from paintLayers using removePaintIdFromLayers',
+      () {
+        final comp = _MyComponent();
+
+        const color = Color(0xFFE5E5E5);
+        const anotherColor = Color(0xFFABABAB);
+        const thirdColor = Color(0xFF123456);
+        comp.setPaint('test', Paint()..color = color);
+        comp.setPaint('anotherTest', Paint()..color = anotherColor);
+        comp.setPaint('thirdTest', Paint()..color = thirdColor);
+        comp.setPaintLayers(['test', 'anotherTest', 'thirdTest']);
+        comp.removePaintIdFromLayers('anotherTest');
+
+        expect(comp.paintLayers[1].color, thirdColor);
+      },
+    );
+
+    test(
+      'use setPaintLayers to set mutliple paintIds in paintLayers',
+      () {
+        final comp = _MyComponent();
+
+        const color = Color(0xFFE5E5E5);
+        const anotherColor = Color(0xFFABABAB);
+        const thirdColor = Color(0xFF123456);
+        comp.setPaint('test', Paint()..color = color);
+        comp.setPaint('anotherTest', Paint()..color = anotherColor);
+        comp.setPaint('thirdTest', Paint()..color = thirdColor);
+
+        comp.setPaintLayers(['thirdTest', 'test']);
 
         expect(
-          () => comp.getPaint(_MyComponentKeys.foreground),
-          throwsArgumentError,
+          (comp.paintLayers[0].color == thirdColor) &&
+              (comp.paintLayers[1].color == color),
+          true,
```

Comment:
```suggestion
          isTrue,
```

Reply:
Thanks for these, I'm not yet very familiar with dart testing